src/unlock/omni_lock.rs

```diff
@@ -228,6 +280,10 @@ impl OmniLockConfig {
         bytes.put(self.id.auth_content.as_ref());
         bytes.put_u8(self.omni_lock_flags.bits);
 
+        if self.admin_config.is_some() {
+            bytes.put(self.admin_config.as_ref().unwrap().rc_type_id.as_bytes());
```

Comment:
Use `.expect("reason")` instead of `.unwrap()`, so that when the function panic we can see the panic message.

Reply:
Is the is_some() check not be able to prevent the panic happen?

Reply:
In this case, the proper way is use `if let Some(config) = self.admin_config.as_ref()`

Reply:
In general, use `.expect()` is always better than `.unwrap()`, since a panic message is helpful in release build.